### PR TITLE
list_menu; Try matching FUN_0202EFE4

### PR DESCRIPTION
**Pull request by PikalaxALT**

### CODE REVIEW

File: arm9/src/list_menu.c [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+    list->fillValue = list->template.fillValue;
+    list->cursorShadowPal = list->template.cursorShadowPal;
+    list->lettersSpacing = list->template.lettersSpacing;
+    list->fontId = list->template.fontId;
+    list->overrideEnabled = FALSE;
+    if (list->template.totalItems < list->template.maxShowed)
+        list->template.maxShowed = list->template.totalItems;
+    ListMenuCursorSetColor(
+        list->cursor,
+//        MakeFontColor(list->template.cursorPal, list->template.cursorShadowPal, list->fillValue)
```
Comment (red031000) [inline arm9/src/list_menu.c:43]: what happened here?

File: arm9/src/list_menu.c [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+
+    return -1;
+}
+
+THUMB_FUNC void ListMenuSetTemplateField(struct ListMenu * list, u32 attr, s32 value)
+{
+    switch (attr)
+    {
+    case 0:
+        list->template.moveCursorFunc = (LM_MOVECURSORFUNC)value;
```
Comment (red031000) [inline arm9/src/list_menu.c:252]: ngl these casts feel really awkward

File: arm9/src/text.c [hunk 3]
```diff
@@ -214,7 +214,7 @@ THUMB_FUNC u16 AddTextPrinter(struct TextPrinterTemplate *printerTemplate, u32 s
         }
         if (speed != 0xff)
         {
-            CopyWindowToVram(printer->printerTemplate.windowId); // CopyWindowToVram?
+            CopyWindowToVram(printer->printerTemplate.window); // CopyWindowToVram?
```
Comment (red031000) [inline arm9/src/text.c:217]: honestly you can remove the CopyWindowToVram comment here

File: arm9/src/unk_0202E29C.c [hunk 4]
```diff
@@ -917,63 +913,49 @@ THUMB_FUNC struct UnkStruct_0202E29C_sub4 *FUN_0202EF44(s32 param0)
...
     {
         GF_ASSERT(param1 < 8);
 
-        MI_CpuCopy8(param0, UNK_021C59E8.unk8->unkC54[param1], 6);
+        MI_CpuCopy8(param0, UNK_021C59F0->unkC54[param1], 6);
     }
 }
 
 #ifdef NONMATCHING
 THUMB_FUNC BOOL FUN_0202EFE4(u8 *param0)
```
Comment (red031000) [inline arm9/src/unk_0202E29C.c:942]: I assume that this func still does not match?

File: include/list_menu.h [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+    /*0x2c*/ u16 cursorPos;
+    /*0x2e*/ u16 itemsAbove;
+    /*0x30*/ u8 unk_30;
+    /*0x31*/ u8 unk_31;
+    /*0x32*/ u8 taskId;
+    /*0x33*/ u8 unk_33;
+    /*0x34*/ u8 heap_id;
+};
+
+THUMB_FUNC struct ListMenu * ListMenuInit(const struct ListMenuTemplate * template, u16 cursorPos, u16 itemsAbove, u32 heap_id);
```
Comment (red031000) [inline include/list_menu.h:73]: remove thumb_func from headers pls

File: include/heap.h [hunk 6]
```diff
@@ -28,7 +28,7 @@ THUMB_FUNC void FUN_02016988();
 void *AllocFromHeap(u32 heap_id, u32 size);
 void *AllocFromHeapAtEnd(u32 heap_id, u32 size);
 void FreeToHeap(void *ptr);
-void FUN_02016A8C(u32 heap_id, void * ptr);
+void FreeToHeapExplicit(u32 heap_id, void * ptr);
 THUMB_FUNC u32 FUN_02016AF8(u32 param0);
```
Comment (red031000) [inline include/heap.h:32]: can you clean this header up too, I know you didn't do it but that would be appreciated

File: include/list_menu_cursor.h [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+
+#include "window.h"
+
+// TODO: Move to its own header, and fill it out
+struct ListMenuCursor
+{
+    // ???
+};
+
+extern struct ListMenuCursor * ListMenuCursorNew(u32 heap_id);
```
Comment (red031000) [inline include/list_menu_cursor.h:12]: extern not needed here? or has the file not been decomped yet?

File: include/window.h [hunk 8]
```diff
@@ -0,0 +1,25 @@
...
+    u8 unk_06;
+    u8 width;
+    u8 height;
+    u8 unk_9;
+    u16 unk_A_0:15;
+    u16 unk_A_15:1;
+    void * unk_C;
+};
+
+extern void FillWindowPixelBuffer(struct Window *, u32);
```
Comment (red031000) [inline include/window.h:18]: again idk if window.c has been decomped but i'm too lazy to find out, if it has then the externs are not necessary

**Review by red031000 — CHANGES_REQUESTED** [review]
I am literally doing this at 5:21 am someone send help

File: arm9/src/list_menu.c (hunk 1 above)
Comment (PikalaxALT) [reply arm9/src/list_menu.c:43]: Using that static inline there does not match yet. But when it does, the commented line will replace the ones below.

File: arm9/src/list_menu.c (hunk 2 above)
Comment (PikalaxALT) [reply arm9/src/list_menu.c:252]: Can't assign an integer to a pointer type without an explicit cast.

File: arm9/src/unk_0202E29C.c (hunk 4 above)
Comment (PikalaxALT) [reply arm9/src/unk_0202E29C.c:942]: Aye. But it's closer.

File: include/list_menu_cursor.h (hunk 7 above)
Comment (PikalaxALT) [reply include/list_menu_cursor.h:12]: Indeed it has not been decomped yet.

File: include/window.h (hunk 8 above)
Comment (PikalaxALT) [reply include/window.h:18]: It has not

File: arm9/src/list_menu.c (hunk 2 above)
Comment (red031000) [reply arm9/src/list_menu.c:252]: I know, it's just the type names feel awkward being in all caps

**Review by red031000 — APPROVED** [review]

**Pull request merged**